libraries/AP_Networking/AP_Networking_port.cpp

```diff
@@ -368,24 +385,14 @@ bool AP_Networking::Port::send_receive(void)
             WITH_SEMAPHORE(sem);
             n = writebuffer->peekbytes(buf, available);
         }
-        if (n > 0) {
-            const auto ret = sock->send(buf, n);
+        if (n > 0 && last_udp_connect_address != 0 && last_udp_connect_port != 0) {
+            const auto ret = sock->sendto(buf, n, last_udp_connect_address, last_udp_connect_port);
```

Comment:
this looks like we will end up using sendto() for tcp client, udp client and tcp server, which doesn't seem right. It only makes sense to do this for UDP server

Reply:
fixed